pyblish_nuke/lib.py

```diff
@@ -56,7 +68,12 @@ def show():
 
     """
 
-    return (_discover_gui() or _show_no_gui)()
+    window = (_discover_gui() or _show_no_gui)()
+
+    if self._dock:
```

Comment:
That's not a bad approach..

If you put a safeguard in there, something about `widget` needing to be an instance of `QWidget` and the window not already having a parent, we could give it a try. I could see that work.